PhysicsTools/NanoAOD/plugins/VertexTableProducer.cc

```diff
@@ -131,6 +139,28 @@ void VertexTableProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSe
   pvTable->addColumnValue<float>(
       "score", pvsScoreProd.get(pvsIn.id(), 0), "main primary vertex score, i.e. sum pt2 of clustered objects", 8);
 
+  float pv_sumpt2 = 0.0;
+  for (size_t i = 0; i < (*pfcIn).size(); i++) {
+    if ((*pfcIn)[i].charge() == 0) {
+      continue;
+    }  // skip neutrals
+    double dz = fabs((*pfcIn)[i].dz((*pvsIn)[0].position()));
+    int include_pfc = 0;
+    if (dz < 0.2)
+      include_pfc = 1;
+    for (size_t j = 1; j < (*pvsIn).size(); j++) {
+      double newdz = fabs((*pfcIn)[i].dz((*pvsIn)[j].position()));
+      if (newdz < dz)
+        include_pfc = 0;
```

Comment:
you probably need to add a "break" statement here, to avoid running over all vertices